client/container_list.go

```diff
@@ -35,7 +37,14 @@ func (cli *Client) ContainerList(ctx context.Context, options types.ContainerLis
 	}
 
 	if options.Filter.Len() > 0 {
-		filterJSON, err := filters.ToParam(options.Filter)
+		filterJSON := ""
+		err := errors.New("")
+		// for daemons older than v1.10, filter must be of the form map[string][]string
+		if cli.version != "" && version.Version(cli.version).LessThan(version.Version("1.10")) {
```

Comment:
Okay, new function makes sense.

I agree that it's a bad idea in general. In Swarm, we can certainly handle the transformation. There's just one thing that is missing then - the `version` in the apiClient needs to be updated from time to time depending on which daemon is being talked to (Swarm will handle this). But is that currently doable in `engine-api`?